gapis/api/transform2/file_log.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transform2
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/gapis/api"
+	"github.com/google/gapid/gapis/config"
+)
+
+type fileLog struct {
+	file *os.File
+}
+
+// NewFileLog returns a Transformer that will log all commands passed through it
```

Comment:
You probably want an interface check here for fileLog implementing the Transform interface.

Reply:
As far as I have seen, compiler does it for you if you want to add the fileLog to a transform slice. Correct me if there are cases that it does not.